ttnlogger/ttn_logger.py

```diff
@@ -101,15 +118,24 @@ def store(self, ttn_message):
             data[field] = value
 
         # Pick up telemetry values from gateway information.
-        gw0 = ttn_message.metadata.gateways[0]
-        data['gw_rssi'] = float(gw0.rssi)
-        data['gw_snr'] = float(gw0.snr)
-        data['gw_latitude'] = float(gw0.latitude)
-        data['gw_longitude'] = float(gw0.longitude)
-        data['gw_altitude'] = float(gw0.altitude)
+        num_gtws = len(ttn_message.metadata.gateways)
+        print('Message received from ' + str(num_gtws) + ' gateway(s)')
+
+        for i in range(num_gtws):
+            gtw_id = ttn_message.metadata.gateways[i].gtw_id
+            print('Gateway ' + str(i+1) + ' ID          : ' + gtw_id)
+            key_rssi = 'gw_' + gtw_id + '_rssi'
+            key_snr  = 'gw_' + gtw_id + '_snr'
+            data[key_rssi] = int(ttn_message.metadata.gateways[i].rssi)
+            data[key_snr] = float(ttn_message.metadata.gateways[i].snr)
+
+        data['sf'] = int(ttn_message.metadata.data_rate.split('BW')[0][2:])
+        data['bw'] = int(ttn_message.metadata.data_rate.split('BW')[1])
```

Comment:
I don't quite see through that yet.

Reply:
Payloads can be received from several gateways at once and each gateway puts the individual signal quality data (RSSI and SNR) into the message. Spreading factor (sf) and bandwidth are determinated by the sending device and thus the same for every receiver.

Reply:
And, on multiple receiving gateways there's no defined order of apperance in the TTN message. Thus, always picking `gw0` will point to random gateways.

Reply:
> On multiple receiving gateways there's no defined order of appearance in the TTN message.

That's sad. But we are keying by `gtw_id` now, right? This contains stable identifiers like `eui-3338333018004700`.

Reply:
> Spreading factor (sf) and bandwidth (bw) are determinated by the sending device and thus equals for every receiver.

So, the last two lines actually decode the
```
"data_rate": "SF7BW125"
```
right?